File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int n=s.length();
        StringBuilder sb= new StringBuilder();
        for(char c: s.toCharArray()){
            if(Character.isLetterOrDigit(c)){
                sb.append(Character.toLowerCase(c));
            }
        }
        int left=0;
        int right=sb.length()-1;
         s=sb.toString();
        while(left<=right){
            if(s.charAt(left)!=s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        st=""
        for i,ch in enumerate(s):
            if ch.isalpha() or ch.isdigit():
                st+=ch.lower()
        
        l,r= 0,len(st)-1

        while(l<=r):
            if st[l]!=st[r]:
                return False
            l+=1
            r-=1
        return True
        
            
